docs/getting-started.md

```diff
@@ -0,0 +1,260 @@
+# Getting Started
+
+In this guide, you'll learn how to deploy, test, and upgrade a contract using ChugSplash.
+
+## Table of Contents
+
+1. [Setup a Foundry project](#1-setup-a-foundry-project)
+2. [Install ChugSplash](#2-install-chugsplash)
+3. [Configure your `foundry.toml` file](#3-configure-your-foundrytoml-file)
+4. [Create a contract](#4-create-a-contract)
+5. [Create a ChugSplash file](#5-create-a-chugsplash-file)
+6. [Create your deployment script](#6-create-your-deployment-script)
+7. [Deploy with ChugSplash](#7-deploy-with-chugsplash)
+8. [Test with ChugSplash](#8-test-with-chugsplash)
+9. [Upgrade with ChugSplash](#9-upgrade-with-chugsplash)
+
+## Prerequisites
+
+The following must be installed on your machine:
+- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+- [Foundry](https://book.getfoundry.sh/getting-started/installation)
+
+You must also have a basic understanding of how to use Foundry. [See here](https://book.getfoundry.sh/getting-started/first-steps) for a brief introduction.
+
+## 1. Setup a Foundry project
+
+If you have an existing foundry project, navigate to it then [skip to step #2](#2-install-chugsplash).
+
+If you're starting a new project, run:
+
+```
+forge init hello_foundry && cd hello_foundry
+```
+
+Then, delete the files that come with the default Foundry project:
+```
+rm src/Counter.sol script/Counter.s.sol test/Counter.t.sol
+```
+
+## 2. Install ChugSplash
+
+In your project root, run:
+
+```
+forge install chugsplash/chugsplash-foundry --no-commit
+```
+
+Then install dependencies by running:
+
+```
+(cd lib/chugsplash-foundry && npm install)
+```
+
+## 3. Configure your `foundry.toml` file
+
+Edit your `foundry.toml` file to include all of the following options. If you leave any of these out, ChugSplash will not work properly.
+
+```
+[profile.default]
+ffi = true
+build_info = true
+extra_output = ['storageLayout']
+force = true
+fs_permissions = [{ access = "read", path = "./"}]
+remappings = [
+  'chugsplash-foundry/=lib/chugsplash-foundry/'
+]
+
+[rpc_endpoints]
+localhost = "http://127.0.0.1:8545"
+```
+
+## 4. Create a contract
+
+We'll now setup a ChugSplash project to deploy a single upgradeable contract, `HelloChugSplash.sol`.
+
+First, define `HelloChugSplash.sol` in your contract source folder (usually, this is `src/`).
+
+Copy and paste its contents:
+```sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+contract HelloChugSplash {
+    uint public number;
+    bool public stored;
+    address public otherStorage;
+    string public storageName;
```

Comment:
It might be nice to reiterate that this will be deployed as an upgradable contract.